posix-mapper/src/main/java/org/opencadc/posix/mapper/web/PosixMapperListener.java

```diff
@@ -0,0 +1,36 @@
+package org.opencadc.posix.mapper.web;
+
+import ca.nrc.cadc.auth.AuthenticationUtil;
+import ca.nrc.cadc.auth.IdentityManager;
+import org.apache.log4j.Logger;
+import org.opencadc.posix.mapper.auth.DelegatingAPIKeyIdentityManager;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+
+
+/**
+ * A bit of an ugly hack to reset the IdentityManager with a delegating one.
+ */
+public class PosixMapperListener implements ServletContextListener {
```

Comment:
This could be done in the InitAction.doInit() and doShutdown() which is intended for this kind of thing.

Yes, this listener mechanism exists, but the cadc-rest one allows you to take advantage of context or endpoint specific strings to make "more local" JNDI keys (if necessary).... it is not necessary in this case because the basic IdentityManager thing is based on java system prop config;  it's also the common place to look for service initialization so **I'd prefer it be in there**. Also, we can then control when this happens relative to other initialization and make use of other config in case that ever matters.

Reply:
Sure.  We currently have two Servlets (one for UID and another for GID).  They both configure an `InitAction` for use now, which means this will be done twice, but it *must* be done only once.  Is it preferable to combine those two Servlets and have a single `GetAction`, or just create a dummy Servlet with an `InitAction` that only does this work?